### PR TITLE
[zh-cn] update Firefox UA string reference

### LINKED ISSUE #18790: firefox agent error
### MDN URL

https://developer.mozilla.org/zh-CN/docs/Web/HTTP/Headers/User-Agent/Firefox

### 与此议题相关的小节或标题？

#Windows

### 哪些信息是错误、无用或不完善的？

Windows NT, WOW64   Mozilla/5.0 (Windows NT x.y; WOW64; rv:10.0) Gecko/20100101 Firefox/10.0

### 你期望看到什么？

Firefox 69版本已经不使用WOW64

### 你有什么支持的链接、参考或引用？

https://www.ghacks.net/2019/06/22/mozilla-omits-cpu-architecture-information-from-firefoxs-user-agent/

### 你还想补充什么内容？

编辑useragent 的不同阶段变化

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `zh-cn/web/http/headers/user-agent/firefox`
* MDN URL: https://developer.mozilla.org/zh-CN/docs/Web/HTTP/Headers/User-Agent/Firefox
* GitHub URL: https://github.com/mdn/translated-content/blob/main/files/zh-cn/web/http/headers/user-agent/firefox/index.md
* Last commit: https://github.com/mdn/translated-content/commit/f873c44624a9ca26f187f0f44e504f5386d55ca3
* Document last modified: 2023-08-31T02:11:59.000Z

</details>

### PR DESCRIPTION
**Pull request by jasonren0403**
### Related issues and pull requests

fixes #18790

### CODE REVIEW

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 1]
```diff
@@ -1,120 +1,170 @@
...
-- _**Mozilla/5.0**_ 表明该浏览器与 Mozilla 相兼容的通用字符串。此字段几乎存在于现今所有浏览器的用户代理之中。
-- **_platform_** 描述了浏览器所运行的本地操作系统平台（例如：Windows，Mac，Linux 或 Android），以及其是否为移动设备。Firefox OS 设备只会简单地表示为"Mobile"；因为它实际上基于 Web。请注意，**_platform_** 的值可包含多个被 ";" 分隔的字段。更多细节与例子可以参考下面内容。
-- **rv:_geckoversion_** 显示了 Gecko 的版本号，（如*"17.0"*）。在最近几个版本的浏览器中，**_geckoversion_** 与 **_firefoxversion_** 一致。
-- **_Gecko/geckotrail_** 标明浏览器基于 Gecko。
-- 在桌面端，_**geckotrail**_ 的值恒为"20100101"。
-- _**Firefox/firefoxversion**_ 标明浏览器是 Firefox，并提供了版本号（如"_17.0"_）。
-- 自从移动版 Firefox 10 开始，**_geckotrail_**与**_firefoxversion_**保持一致。
+- `Mozilla/5.0` 是表示浏览器与 Mozilla 兼容的通用标记，目前几乎所有浏览器都使用该标记。
+- `platform` 描述浏览器运行的本地平台（如 Windows、Mac、Linux 或 Android），以及是否是手机。Firefox OS 手机显示“`Mobile`”；web 就是平台。请注意，`platform` 可以由多个“;”分隔的标记组成。详情和示例请参阅下文。
+- `rv:geckoversion` 表示 Gecko 的发布版本（如“`17.0`”）。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:20]:
```suggestion
- `rv:geckoversion` 表示 Gecko 的发行版本（如“`17.0`”）。
```

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 2]
```diff
@@ -1,120 +1,170 @@
...
+- `Gecko/geckotrail` 表示浏览器基于 Gecko。
+- 在桌面上，`geckotrail` 是固定字符串“`20100101`”。
+- `Firefox/firefoxversion` 表示浏览器是 Firefox，并提供版本（如“`17.0`”）。
+- 从移动版 Firefox 10 开始，`geckotrail` 与 `firefoxversion` 相同。
 
-> **备注：** （如果你必须检测浏览器引擎，而不是去做特征检测的话）推荐使用"_Gecko_"与"_rv:_"字符串来检测基于 Gecko 的浏览器。因为一些浏览器的 UA 中也包含有"_like Gecko_"字段。
+> **备注：** 嗅探基于 Gecko 的浏览器（如果你*必须*嗅探浏览器引擎而不是使用特性检测）的推荐方法是通过“`Gecko`”和“`rv:`”字符串的存在，因为某些其他浏览器包含"`like Gecko`"标记。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:26]: Free translation would be better here

```suggestion
> **备注：** 嗅探基于 Gecko 的浏览器（如果你*必须*嗅探浏览器引擎而不是使用特性检测）的推荐方法是检测“`Gecko`”和“`rv:`”字符串是否存在，因为某些其他浏览器包含"`like Gecko`"标记。
```

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 3]
```diff
@@ -1,120 +1,170 @@
...
 对于其他基于 Gecko 的产品，UA 字符串会符合以下两个规则中的一个，除了下面的描述外，其他字段与上面描述的意义相同。
 
-**Mozilla/5.0 (_platform_; rv:_geckoversion_) Gecko/_geckotrail_ _appname_/_appversion_**
-**Mozilla/5.0 (_platform_; rv:_geckoversion_) Gecko/_geckotrail_ Firefox/_firefoxversion_ _appname_/_appversion_**
+`Mozilla/5.0 (platform; rv:geckoversion) Gecko/geckotrail appname/appversion`
+`Mozilla/5.0 (platform; rv:geckoversion) Gecko/geckotrail Firefox/firefoxversion appname/appversion`
 
-- **_appname/appversion_** 标明了应用名与版本，举个例子，这个字段可以是"_Camino/2.1.1_"或"_SeaMonkey/2.7.1_"。
-- _**Firefox/firefoxversion**_ 是一个可选的兼容字段，一些基于 Gecko 的浏览器选择加入，以使那些支持 Firefox 的网页进行最大的兼容。_**firefoxversion**_ 应与其提供的 Gecko 版本对应的 Firefox 版本相同。一些基于 Gecko 的浏览器可能未加入此此段，因此，在检测的时候，应当使用"Gecko"而非"Firefox"。是否展示这个字段由 about:config 中*"general.useragent.compatMode.firefox"*项设定的布尔值控制。
+- `appname/appversion` 标明了应用名与版本，举个例子，这个字段可以是“Camino/2.1.1”或“SeaMonkey/2.7.1”。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:33]:
```suggestion
- `appname/appversion` 标明了应用名与版本，举个例子，它可以是“`Camino/2.1.1`”或“`SeaMonkey/2.7.1`”。
```

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 4]
```diff
@@ -1,120 +1,170 @@
...
+- `Firefox/firefoxversion` 是一个可选的兼容字段，一些基于 Gecko 的浏览器选择加入，以使那些支持 Firefox 的网页进行最大的兼容。`firefoxversion` 应与其提供的 Gecko 版本对应的 Firefox 版本相同。一些基于 Gecko 的浏览器可能不会使用此标记，因此，在检测的时候，应当使用 Gecko 而非 Firefox。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:34]:
```suggestion
- `Firefox/firefoxversion` 是一个可选的兼容标记，一些基于 Gecko 的浏览器选择加入，以达到对支持 Firefox 的网页的最大兼容。`firefoxversion` 应与其提供的 Gecko 版本对应的 Firefox 版本相同。一些基于 Gecko 的浏览器可能不会使用此标记，因此，在检测的时候，应当使用 Gecko 而非 Firefox。
```

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 5]
```diff
@@ -1,120 +1,170 @@
...
 
-- _**Mozilla/5.0**_ 表明该浏览器与 Mozilla 相兼容的通用字符串。此字段几乎存在于现今所有浏览器的用户代理之中。
-- **_platform_** 描述了浏览器所运行的本地操作系统平台（例如：Windows，Mac，Linux 或 Android），以及其是否为移动设备。Firefox OS 设备只会简单地表示为"Mobile"；因为它实际上基于 Web。请注意，**_platform_** 的值可包含多个被 ";" 分隔的字段。更多细节与例子可以参考下面内容。
-- **rv:_geckoversion_** 显示了 Gecko 的版本号，（如*"17.0"*）。在最近几个版本的浏览器中，**_geckoversion_** 与 **_firefoxversion_** 一致。
-- **_Gecko/geckotrail_** 标明浏览器基于 Gecko。
-- 在桌面端，_**geckotrail**_ 的值恒为"20100101"。
-- _**Firefox/firefoxversion**_ 标明浏览器是 Firefox，并提供了版本号（如"_17.0"_）。
-- 自从移动版 Firefox 10 开始，**_geckotrail_**与**_firefoxversion_**保持一致。
+- `Mozilla/5.0` 是表示浏览器与 Mozilla 兼容的通用标记，目前几乎所有浏览器都使用该标记。
+- `platform` 描述浏览器运行的本地平台（如 Windows、Mac、Linux 或 Android），以及是否是手机。Firefox OS 手机显示“`Mobile`”；web 就是平台。请注意，`platform` 可以由多个“;”分隔的标记组成。详情和示例请参阅下文。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:19]: suggestion, see https://zh.wikipedia.org/wiki/Firefox_OS

```suggestion
- `platform` 描述浏览器运行的本地平台（如 Windows、Mac、Linux 或 Android），以及是否是手机。Firefox OS 手机显示“`Mobile`”；web 就是它的平台。请注意，`platform` 可以由多个“;”分隔的标记组成。详情和示例请参阅下文。
```

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 6]
```diff
@@ -1,120 +1,170 @@
...
 ```
 
-> **备注：** 上述版本号并不对应当前版本，请避免基于此例子中的版本号。
+> **备注：** 版本号无关。请避免据此推断材料。
 
-推荐的判断设备尺寸的方法是使用 CSS 媒体查询 (CSS Media Query)。但是，当你需要使用 UA 判断用户是否使用移动尺寸的设备，请查找"**Mobi**"（以兼容使用"Mobi"的 Opera 移动浏览器）。**不要** 假定"Android"字符串与设备尺寸存在关联。这样的话，在“Firefox 在其他操作系统的移动尺寸设备上运行”或“使用 Android 的笔记本电脑”的情形下，你的代码才能正确运行。此外，请使用触摸检测以分辨设备是否支持触摸，不要使用"Mobi"或"Tablet"，因为也有很多触摸设备并非移动设备或平板。
+将内容定位到设备外形尺寸的首选方法是使用 CSS 媒体查询。但是，如果你使用 UA 嗅探将内容定位到设备外形，请在手机外形中查找 **Mobi**（包括使用 "Mobi "的 Opera Mobile），而**不要**假设“Android”与设备外形之间存在任何关联。这样，如果 Firefox 浏览器采用其他手机/平板电脑操作系统，或者笔记本电脑采用 Android 系统，你的代码也能正常工作。此外，请使用触摸检测来查找触摸设备，而不是查找“Mobi”或“Tablet”，因为可能存在不是平板电脑的触摸设备。
 
-> **备注：** Firefox OS 设备的 UA 中不会有标明操作系统的字段，比如"Mozilla/5.0 (Mobile; rv:15.0) Gecko/15.0 Firefox/15.0"。因为其平台实为 Web。
+> **备注：** Firefox OS 设备在标识自己时不标注任何操作系统，例如“Mozilla/5.0 (Mobile; rv:15.0) Gecko/15.0 Firefox/15.0”，web 就是平台。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:51]:
```suggestion
> **备注：** Firefox OS 设备在标识自己时不标注任何操作系统，例如“Mozilla/5.0 (Mobile; rv:15.0) Gecko/15.0 Firefox/15.0”。web 就是它的平台。
```

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 7]
```diff
@@ -1,120 +1,170 @@
...
-| ---------------------------------------------- | ---------------------------------------------------------------------------------- |
-| Linux desktop, i686                            | Mozilla/5.0 (X11; Linux i686; rv:10.0) Gecko/20100101 Firefox/10.0                 |
-| Linux desktop, x86_64                          | Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20100101 Firefox/10.0               |
-| Linux desktop, i686 running on x86_64          | Mozilla/5.0 (X11; Linux i686 on x86_64; rv:10.0) Gecko/20100101 Firefox/10.0       |
-| Nokia N900 Linux mobile, on the Fennec browser | Mozilla/5.0 (Maemo; Linux armv7l; rv:10.0) Gecko/20100101 Firefox/10.0 Fennec/10.0 |
+| Linux 版本                        | Gecko 用户代理字符串                                                 |
+| --------------------------------- | -------------------------------------------------------------------- |
+| 在 i686 CPU 运行的 Linux 桌面版   | Mozilla/5.0 (X11; Linux i686; rv:10.0) Gecko/20100101 Firefox/10.0   |
+| 在 x86_64 CPU 运行的 Linux 桌面版 | Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20100101 Firefox/10.0 |
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:64]: 在 xxx CPU 上运行的 xxx

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 8]
```diff
@@ -1,120 +1,170 @@
...
-从 41 版本开始，Firefox Android 会在 _platform_ 字段中包含 Android 版本号。为增加兼容性，如果浏览器在 Android 4 及以下版本的 Android 设备上运行，此字段会使用 4.4，Android 4.4 及以上版本的则会使用当前版本号。请注意，目前发布至不同版本的 Android 设备的 Gecko，是拥有相同兼容性的同一个 Gecko 的不同架构（ARMv7、x86、x86_64）编译版。
+从版本 41 开始，Android 版 Firefox 浏览器将把 Android 版本作为 _platform_ 标记的一部分。为提高互操作性，如果浏览器运行的版本低于 4，它将报告 4.4。Android 4 及以上版本将准确报告版本。请注意，所有版本的 Android 系统都使用具有相同功能的 Gecko。
 
-| 设备类型 | Gecko User Agent 值                                                |
+| 设备类型 | Gecko 用户代理字符串                                               |
 | -------- | ------------------------------------------------------------------ |
 | Phone    | Mozilla/5.0 (Android 4.4; Mobile; rv:41.0) Gecko/41.0 Firefox/41.0 |
 | Tablet   | Mozilla/5.0 (Android 4.4; Tablet; rv:41.0) Gecko/41.0 Firefox/41.0 |
 
+## Focus for Android
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:82]: how about:

```suggestion
## Android 版 Focus
```

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 9]
```diff
@@ -1,120 +1,170 @@
...
-| TV              | Mozilla/5.0 (TV; rv:44.0) Gecko/44.0 Firefox/44.0                 |
-| Device-specific | Mozilla/5.0 (Mobile; _**nnnn;**_ rv:26.0) Gecko/26.0 Firefox/26.0 |
+| 设备类型 | Gecko 用户代理字符串                                              |
+| -------- | ----------------------------------------------------------------- |
+| Phone    | Mozilla/5.0 (Mobile; rv:26.0) Gecko/26.0 Firefox/26.0             |
+| Tablet   | Mozilla/5.0 (Tablet; rv:26.0) Gecko/26.0 Firefox/26.0             |
+| TV       | Mozilla/5.0 (TV; rv:44.0) Gecko/44.0 Firefox/44.0                 |
+| 设备特定 | Mozilla/5.0 (Mobile; **_nnnn;_** rv:26.0) Gecko/26.0 Firefox/26.0 |
 
 ### 具有设备特征的 User Agent 值
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:153]:
```suggestion
### 具有设备特征的用户代理字符串
```

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 10]
```diff
@@ -1,120 +1,170 @@
...
 
-尽管 Mozilla **强烈反对**，然而一些手机制造商仍然会在其设备的 User Agent 值中包含标明设备 ID 的字段。在这种情况下，Firefox OS 的 User Agent 会与上表中所示的具有设备特征 (Device-specific) 的字符串保持一致，其中**_nnnn;_**代表设备制造商的设备代码（参见[引导](https://wiki.mozilla.org/B2G/User_Agent/Device_Model_Inclusion_Requirements)）。其中一些是以下几个形式： "**NexusOne;**"，"**ZTEOpen;**"，"**Open C;**"（另请注意，同样不鼓励在其中加入空格）。我们提供这些信息以辅助你设定 User Agent 判断逻辑，但 Mozilla 不鼓励从 User Agent 值中检测设备 ID。
+尽管 Mozilla **强烈反对**，然而一些手机制造商仍然会在其设备的 User Agent 值中包含标明设备 ID 的字段。在这种情况下，Firefox OS 的 User Agent 会与上表中所示的具有设备特征的字符串保持一致，其中 **_nnnn;_** 代表设备制造商的设备代码（参见[指南](https://wiki.mozilla.org/B2G/User_Agent/Device_Model_Inclusion_Requirements)）。其中一些是以下几个形式：“**NexusOne;**”、“**ZTEOpen;**”、“**Open C;**”（另请注意，同样不鼓励在其中加入空格）。我们提供这些信息是为了协助你的 UA 检测逻辑，但 Mozilla 不鼓励在 UA 字符串中检测设备 ID。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:155]:
```suggestion
尽管 Mozilla **强烈反对**，然而一些手机制造商仍然会在其设备的 UA 字符串中包含标明设备 ID 的字段。在这种情况下，Firefox OS 的 UA 字符串会与上表中所示的具有设备特征的字符串保持一致，其中 **_nnnn;_** 代表设备制造商的设备代码（参见[指南](https://wiki.mozilla.org/B2G/User_Agent/Device_Model_Inclusion_Requirements)）。其中一些是以下几个形式：“**NexusOne;**”、“**ZTEOpen;**”、“**Open C;**”（注意，同样不鼓励在其中加入空格）。我们提供这些信息是为了协助你的 UA 检测逻辑，但 Mozilla 不鼓励检测 UA 字符串中的设备 ID。
```

File: files/zh-cn/web/http/headers/user-agent/firefox/index.md [hunk 11]
```diff
@@ -1,120 +1,170 @@
...
+```js
+/mobi/i;
 ```
 
 其中 `i` 为大小写敏感的标记，而 `mobi` 则将匹配所有的移动设备。
 
 ### Firefox OS 版本号
 
-由于 Firefox OS 的版本号未被包含于 User Agent 值中，我们可以通过 User Agent 中的 Gecko 版本号来推断其版本信息。
+由于 Firefox OS 的版本号未被包含于 User Agent 值中，我们可以通过用户代理中的 Gecko 版本号来推断其版本信息。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/user-agent/firefox/index.md:167]:
```suggestion
由于 Firefox OS 的版本号未被包含于 UA 字符串中，我们可以通过 UA 字符串中的 Gecko 版本号来推断其版本信息。
```

**Review by yin1999 — COMMENTED** [review]
some suggestions:

**Review by JasonLamv-t — COMMENTED** [review]
No addtional comments. LGTM.

**Review by yin1999 — APPROVED** [review]
Thanks @jasonren0403

**Pull request merged**